wicket/src/ui/panes/overview.rs

```diff
@@ -439,3 +444,429 @@ impl Control for InventoryView {
+                            }
+                            None => Span::styled("Unknown", warn_style),
+                        },
+                    ]
+                    .into(),
+                );
+                if let Some(caboose) =
+                    sp.rot().and_then(|r| r.caboose_b.as_ref())
+                {
+                    append_caboose(&mut spans, caboose);
+                } else {
+                    spans.push(
+                        vec![
+                            nest_bullet(),
+                            Span::styled("No further information", warn_style),
+                        ]
+                        .into(),
+                    );
+                }
+            }
+            RotState::CommunicationFailed { message } => {
+                spans.push(label.into());
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled(
+                            format!("Communication Failure: {message}"),
+                            bad_style,
+                        ),
+                    ]
+                    .into(),
+                );
+            }
+        }
+    } else {
+        label.push(Span::styled("Not available", bad_style));
+        spans.push(label.into());
+    }
+
+    // blank line separator
+    spans.push(Spans::default());
+
+    // Describe all components.
+    // TODO-correctness: component information will change with the IPCC /
+    // topology work that is ongoing; show our current stand-in for now.
+    let mut label = vec![Span::styled("Components: ", label_style)];
+    let components = sp.components();
+    if components.is_empty() {
+        label.push(Span::styled("Not available", bad_style));
+        spans.push(label.into());
+    } else {
+        spans.push(label.into());
+        for component_info in components {
+            spans.push(
+                vec![
+                    bullet(),
+                    Span::styled(
+                        format!("{:?}", component_info.presence),
+                        match component_info.presence {
+                            SpComponentPresence::Present => ok_style,
+                            SpComponentPresence::Timeout
+                            | SpComponentPresence::Unavailable => warn_style,
+                            SpComponentPresence::NotPresent
+                            | SpComponentPresence::Failed
+                            | SpComponentPresence::Error => bad_style,
+                        },
+                    ),
+                    Span::styled(
+                        format!(
+                            " {} ({})",
+                            component_info.description, component_info.device
+                        ),
+                        label_style,
+                    ),
+                ]
+                .into(),
+            );
+        }
+    }
+
+    Text::from(spans)
```

Comment:
Does this get wrapped today? If not, then it's worth adding wrapping at some point.

Reply:
Added in 3b1103ae:

![it_wraps](https://github.com/oxidecomputer/omicron/assets/1435635/d04aecaa-de05-427a-8ed0-92109d1f5652)